### PR TITLE
fix: make component testing windows compatible

### LINKED ISSUE #15842: [7.0.0] Doesn't display list of specs in Cypress Component Test Runner
Hello Cypress team,

I have a problem with displaying spec files in new  Cypress Component Test Runner even if I use this example of component testing - `https://github.com/lmiller1990/cypress-react-template`
Also there is an error in terminal after `yarn cypress open-ct` started

### Current behavior
Screenshot_2
Screenshot_1
Screenshot_3

### Desired behavior
Should display spec files in new  Cypress Component Test Runner and should not be an error  in terminal

### Test code to reproduce
`https://github.com/lmiller1990/cypress-react-template`

### Versions

Cypress version - 7.0.0, 
Browser Chrome Version - 89.0.4389.114 
Operating System - windows10
Node version - v15.13.0

### PR DESCRIPTION
**Pull request by elevatebart**
- Closes #15842

### User facing changelog
Repair the component testing build for windows

### Additional details
- Forked https://github.com/liximomo/lazy-compile-webpack-plugin to make it windows compatible (following liximomo's advice of the original PR)
- Made relative paths of specs on windows use forward slash instead of backslash for POSIX compatible paths. Allowing the runner-ct to parse them on the front-end into directories.

### How has the user experience changed?
Now Component Testing works on windows

### CODE REVIEW

File: npm/vue/${PWD}/cypress/plugins/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/// <reference types="cypress" />
+// ***********************************************************
```
Comment (lmiller1990) [inline npm/vue/${PWD}/cypress/plugins/index.js:2]: Looks like this file was accidentally created?

**Review by lmiller1990 — COMMENTED** [review]
One comment... don't we have some way to run windows on CI? 🤔

File: npm/vue/${PWD}/cypress/plugins/index.js (hunk 1 above)
Comment (elevatebart) [reply npm/vue/${PWD}/cypress/plugins/index.js:2]: yes... oops

**Comment by elevatebart** [comment]
@lmiller1990 I believe we might be able to do that on AppVeyor.
CircleCI recently added it too. 
I would still merge this without a test and figure out a way to test it in a future PR

**Comment by lmiller1990** [comment]
@JessicaSachs @elevatebart this should be merged right, not closed?

**Review by lmiller1990 — APPROVED** [review]

**Pull request merged**